src/libsystemd/sd-journal/catalog.c

```diff
@@ -169,8 +170,9 @@ static int finish_item(
                 if (!combined)
                         return log_oom();
 
-                if (ordered_hashmap_update(h, i, combined) < 0)
-                        return log_oom();
+                r = ordered_hashmap_update(h, i, combined);
+                if (r < 0)
+                        return r;
```

Comment:
mm, this function so far logged about all errors, now it won't log for these anymore.

if you rpropagate the original error then please use `return log_error_errno(r, …)`...

(or alternative convert the whole function that it doesn't log on its own anymore but leaves that to the caller, up to you)